### DISCUSSION TITLE
Lagged regressor having only positive values

### DISCUSSION THREAD

**Discussion by Miky14880** [ourownstory/neural_prophet#620] [Q&A - get help using NeuralProphet]
Hello all,
first of all, hats off for the super job done on this package!
Here is my question: I am currently using NP with a lagged regressor that has by def only positive values (mortgage rate).
Nevertheless, when I check the regressor columns for the forecasted steps, I see negative values.
Am I doing something wrong or missing something? Is there any way to limit the regressors to positive values only?

Any suggestion is welcome, thanks in advance!
Regards
Michela

**Comment by ourownstory** [comment]
Hi @Miky14880 
While the regressor may be positive overall, it may still have a negative (speak lowering) effect on the overall forecast. That is what the forecast component is showing you.

**Reply by ourownstory** [reply]
e.g. a high mortgage rate may have a negative effect on volume of home sales.
